### PR TITLE
Enable eslint and fix outstanding lint issues.

### PR DESCRIPTION
**Pull request by prestomation**
*Issue #, if available:*

*Description of changes:*

1. Added a root `lint` command to run eslint and `lint:fix` command.
2. The eslint config from the original repo was hooked into webpack. We now have multiple webpack builds, so created .eslintrc in each package to point at their own webpack. This is required for the tests to lint correctly.
3. Add ignores for the documentation and dist directories to eslint
4. Removed a babel-parse config from eslint that was breaking things for some unknown reason
5. Fixed all the outstanding lint issues, including using `npm run lint:fix` and manually fixing what was left.
6. Add a new action to run the build command on pull requests, which will build/test/lint

I tested this by removing some of the existing eslint ignores and seeing that `npm run build` failed from the root

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kris-work — APPROVED** [review]
I've left a couple questions for you. But if you decide no changes are warranted I've marked the PR as approved.

File: .eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
```
[the remark this answers is not in the document]
Comment (prestomation) [reply .eslintrc.json:1]: thanks for pointing these out, they can be removed

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Run build/tests/lint on pull requests
+
+on:
+  pull_request:
+
+# By default the karma test runners use the karma 'Chrome' runner
+# This is great when running locally because the browser pops up and you get to see what it does
+# This doesn't work in Actions, however, so we set this env var to force the Headless runner
+env:
+  TEST_BROWSERS: ChromeHeadless
```
Comment (runpiw) [inline .github/workflows/pull-request.yml:10]: Are we sure all tests work with headless browser?

Comment (prestomation) [reply .github/workflows/pull-request.yml:10]: Looks like they do: You can verify in the checks on this PR.

**Review by runpiw — APPROVED** [review]

**Pull request merged**